packages/react-native-reanimated/src/errors.ts

```diff
@@ -1,6 +1,35 @@
 'use strict';
 import type { WorkletStackDetails } from './commonTypes';
 
+type ReanimatedError = Error & 'ReanimatedError'; // signed type
+
+interface ReanimatedErrorConstructor extends Error {
+  new (message?: string): ReanimatedError;
+  (message?: string): ReanimatedError;
+  readonly prototype: ReanimatedError;
+}
+
+const ReanimatedErrorConstructor: ReanimatedErrorConstructor =
+  function ReanimatedError(message?: string) {
+    'worklet';
+    const prefix = '[Reanimated]';
+    const errorInstance = new Error(message ? `${prefix} ${message}` : prefix);
+    errorInstance.name = 'ReanimatedError';
+    return errorInstance;
+  } as ReanimatedErrorConstructor;
+
+export { ReanimatedErrorConstructor as ReanimatedError };
+
+/**
+ * Registers ReanimatedError constructor in global scope.
+ * Used to add ReanimatedError in the worklet runtime.
+ */
+export function registerReanimatedError() {
+  'worklet';
+  (global as Record<string, unknown>).ReanimatedError =
+    ReanimatedErrorConstructor;
+}
```

Comment:
How about such addition here:

```ts
if(!_WORKLET) {
throw new Error(...);
}
```

Reply:
I think this will be useful. I will add this change.